### DISCUSSION TITLE
Welcome to uncrustify Discussions!

### DISCUSSION THREAD

**Discussion by gmaurel** [uncrustify/uncrustify#3255] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by micheleCTDE** [comment]
Hi Guy,
thanks for enabling this space.

My name is Michele Calgaro, I am one of the core developers of the Trinity Desktop Environment and we plan to use uncrustify on our code base at some point, hence my involvement in this community.

**Comment by MikeGM2017** [comment]
Hi!

I am trying to use uncrustify, but I have a question:
int foo() { - I can not find such variant:
sp_func_def_paren - exists, but sp_after_func_def_paren - no?

Now I can see only:
int foo(){

**Reply by micheleCTDE** [reply]
Hi Mike, can you better elaborate what you are trying to achieve? Maybe an example "input" + "expected output" would help.

**Reply by MikeGM2017** [reply]
I use linux-indent_V02.cfg (modified linux-indent.cfg):

diff linux-indent.cfg linux-indent_V02.cfg 
32c32
< nl_fdef_brace			= add		# "int foo() {" vs "int foo()\n{"
---
> nl_fdef_brace			= remove	# "int foo() {" vs "int foo()\n{"

Input foo.cpp:

void foo(){}

Output:

./uncrustify -c linux-indent_V02.cfg -f foo.cpp
do_source_file(1535): Parsing: foo.cpp as language CPP
void foo(){
}

Desirable output:

void foo() {
}

(additional space between "foo()" and "{")

**Reply by micheleCTDE** [reply]
you need to set the options for spacing. They start with `sp`, not `nl`
For what you are trying to do, there are a number of `sp_func_*` options available.

**Reply by MikeGM2017** [reply]
Sorry, but I need flag "sp_after_fdef". It is absent.

**Comment by HazemKurdy** [comment]
hey All,
I got a couple of question :
1- is there a method to have two or more configuration files (for example one for c and one for c++)
2- is there a way to add regular expression instead of file/folder name in IgnoreList
thanks

**Comment by guy-maurel** [comment]
1. yes
2. no
